kernel/src/process.rs

```diff
@@ -197,6 +207,46 @@ pub trait Process {
     /// kernel-internal errors.
     fn enqueue_task(&self, task: Task) -> Result<(), ErrorCode>;
 
+    /// Enqueue a `Task` to execute the init function of the process.
+    /// The process must be in the the `Terminated` or `Unstarted`
+    /// state, and invoking this method transitions it to the
+    /// `Yielded` state before enqueuing the task. This is the only
+    /// method that transitions a process from `Terminated` or
+    /// `Unstarted` to `Yielded`. Because starting a process has
+    /// security implications (e.g., that every running process has a
+    /// unique application identifier), this method requires a
+    /// Capability.
+    fn enqueue_init_task(
+        &self,
+        cap: &dyn capabilities::ProcessInitCapability,
+    ) -> Result<(), ErrorCode>;
+
+    /// Transition a loaded but unchecked process into the `Unstarted`
+    /// state so it can run. Returns an error if the process was not
+    /// in the `Unchecked` state. The `credentials` argument is `None`
+    /// if all credentials are Pass; if a credential is Accept, it is
+    /// passed in `credentials`. If `credentials` is `None`,
+    /// `short_app_id` is `None`. If `credentials` is not `None`, The
+    /// `short_app_id` argument is the short ID (if there is one)
+    /// generated by the process checking policy. Transitioning a
+    /// process to `Unstarted` has security implications because it
+    /// means this application binary is permitted to run on the
+    /// system. This method therfore requires a Capability.
+    fn mark_credentials_pass(
+        &self,
+        credentials: Option<TbfFooterV2Credentials>,
+        short_app_id: Option<ShortID>,
+        capability: &dyn capabilities::ProcessApprovalCapability,
+    ) -> Result<(), ErrorCode>;
```

Comment:
The user-correct coupling between `credentials` and `short_app_id` feels odd here. Why pass `short_app_id` at all, why not simply have the internals of `mark_credentials_pass` call `self.short_app_id()` to get the id?

Reply:
The idea is that the short ID shouldn't be filled in until credentials are resolved. Calling `self.short_app_id()` within `mark_credentials_pass` implies it has already been set, and so the credentials chosen to pass the process are already known.